sirepo/auth/__init__.py

```diff
@@ -164,6 +166,14 @@ def init_mock(uid=None, sim_type=None):
         login(sirepo.auth.guest, is_mock=True)
 
 
+def is_premium_user(uid):
+    with auth_db.thread_lock:
+        return auth_db.UserRole.search_by(
```

Comment:
`check_user_has_role` should share this code. It's subtle, but important. I note elsewhere that this should always be on auth_user.